code/datums/gamemode/role/time_agent.dm

```diff
@@ -0,0 +1,165 @@
+	var/list/objects_to_delete = list()
+
+
+/datum/role/time_agent/Greet(var/greeting,var/custom)
+	if(!greeting)
+		return
+
+	var/icon/logo = icon('icons/logos.dmi', logo_state)
+	switch(greeting)
+		if (GREET_CUSTOM)
+			to_chat(antag.current, "<img src='data:image/png;base64,[icon2base64(logo)]' style='position: relative; top: 10;'/> <span class='danger'>[custom]</span>")
+		else
+			to_chat(antag.current, "<img src='data:image/png;base64,[icon2base64(logo)]' style='position: relative; top: 10;'/> <span class='danger'>You are a Time Agent.<br>Specifically you are a scientist by the name of John Beckett, having discovered a method to travel through time, and becoming lost to it. <br>\
+			Now, you are forced to take responsibility for maintaining the time stream by the mysterious 'Time Agency'.<br>\
+			You only have a limited amount of time before this timeline is deemed lost, in which case you will be forcibly extracted and the mission considered a failure.<br>\
+			Locate certain time-sensitive objects scattered around the station, so as to locate the time anomaly and use it for extraction.<br>\
+			This may not be the first time you visit this timeline, and it may not be the last.</span>")
+
+	to_chat(antag.current, "<span class='danger'>Remember that the items you are provided with are largely non-expendable. Try not to lose them, especially the jump charge, as it is your ticket home.</span>")
+
+
+/datum/role/time_agent/ForgeObjectives()
+	AppendObjective(/datum/objective/target/locate/random)
+	if(prob(30))
+		AppendObjective(/datum/objective/target/assassinate)
+	AppendObjective(/datum/objective/freeform/aid)
+
+
+/datum/role/time_agent/process()
+	var/list/datum/objective/jecties = objectives.GetObjectives()
+	if(!jecties.len || locate(/datum/objective/time_agent_extract) in objectives.GetObjectives())
+		return //Not set up yet
+	var/finished = TRUE
+	for(var/datum/objective/O in objectives.GetObjectives())
+		if(!O.IsFulfilled())
+			finished = FALSE
+			break
+	if(finished)
+		to_chat(antag.current, "<span class = 'notice'>Objectives complete. Triangulating anomaly location.</span>")
+		AppendObjective(/datum/objective/time_agent_extract)
+
+/datum/role/time_agent/OnPostSetup()
+	.=..()
+	if(istype(antag.current, /mob/living/carbon/human))
+		equip_time_agent(antag.current)
+		spawn_rand_maintenance(antag.current)
+		spawn()
+			showrift(antag.current,1)
+
+
+/datum/role/time_agent/proc/extract()
+	var/mob/living/carbon/human/H = antag.current
+	H.drop_all()
+	showrift(H,1)
+	qdel(H)
+	for(var/i in objects_to_delete)
+		objects_to_delete.Remove(i)
+		qdel(i)
+	increment_threat(rand(5,10))
+
+/obj/item/device/chronocapture
+	name = "chronocapture device"
+	desc = "Used to confirm that everything is where it should be."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "polaroid"
+	item_state = "polaroid"
+	w_class = W_CLASS_SMALL
+	var/triggered = FALSE
+
+/obj/item/device/chronocapture/afterattack(atom/target, mob/user)
+	if(!triggered)
```

Comment:
I would format this block slightly differently to avoid rightward creep

```
/obj/item/device/chronocapture/afterattack(atom/target, mob/user)
	var/datum/role/R = user.mind.GetRole(TIMEAGENT)
	if(triggered || !istype(R))
		return
	triggered = TRUE
	playsound(loc, "polaroid", 75, 1, -3)
	spawn(3 SECONDS)
		triggered = FALSE
	var/datum/objective/target/locate/L = locate() in R.objectives.GetObjectives()
	if(L)
		L.check(view(target,2))
```